### DISCUSSION TITLE
Customizing Search Results Webpart to Filter Based on Document Metadata Tags

### DISCUSSION THREAD

**Discussion by Tishaay** [microsoft-search/pnp-modern-search#3989] [General]
I am in the process of customizing the Search Results Webpart on one of our SharePoint sites to improve the accuracy of search outcomes. Our document library is structured with metadata tags, and we want to configure the search functionality to display only documents or files tagged with the relevant categories when users input keywords in the search box. In cases where the entered keywords do not align with these categories (e.g., document content such as ingredients or other non-tagged metadata), we would prefer the search to return no results, ensuring that only the most relevant, tagged documents are shown.

To achieve this, I attempted to add new managed properties to the search collection. However, when trying to select those properties in the Search Results Webpart configuration, I noticed that the search schema changes were not reflected.

Could you advise if it is possible to filter the webpart results strictly based on this condition, and what steps may be necessary to resolve the issue with the search schema not updating?

**Comment by kasperbolarsen** [comment]
First of all: MS has announced that they had an issue where new columns wasn't picked up by the search index, it should be resolve now.

To me it sounds like you are looking for a search query like: RefinableString01:{searchTerms} OR RefinableString02:{searchTerms} OR RefinableString03:{searchTerms} 
That would ensure that the search text must be a value in one of the metadata fields. However, this also means that you will NOT be able to search the content of the files, so I would personally not recommed this approach, but would look into using XRANK to boost the documents where the search text matches one of the metadata values

**Reply by Tishaay** [reply]
Hi, I am not familiar with XRANK. How can we configure XRANK in our search results?

**Reply by kasperbolarsen** [reply]
If you search in this repository there is a number of similar questions as XRANK isn't as commonly known as it deserves. This is a good primer, https://searchexplained.com/how-to-sort-search-results-by-using-query-rules/

**Reply by Tishaay** [reply]
In configuring the search results query template, I'm using the following structure:
RefinableString01: {inputQueryText}
Path: URL of the document library

Is this the correct approach for setting up the query to ensure only documents tagged with specific metadata are shown in the search results?
